### PR TITLE
Adds query and component for fetching todos from local-schema

### LINKED ISSUE #3: Get All Todos Query
**Acceptance Criteria**
 - [ ] Create a GetTodo Component
 - [ ] Fetch all the Todos
 - [ ] Render all the Todo items

### PR DESCRIPTION
**Pull request by SiddharthSham**
Resolves #3

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,14 +1,20 @@
-import React, { useState, useEffect } from "react";
+import React, { Suspense } from "react";
```
Comment (jevakallio) [inline src/App.js:1]: Ooh! Exciting! 👀

File: src/data/schema.graphql [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+type Root {
+    viewer: User
+    node(id: ID!): Node
+}
+
+type User implements Node {
+    id: ID!
+    todos(status: String ="any", after: String, before: String, first: Int, last: Int): TodoConnection
```
Comment (jevakallio) [inline src/data/schema.graphql:13]: Is this hand written or autogenerated? If it's hand written, can we fix the weird spacing here:
```suggestion
    todos(status: String = "any", after: String, before: String, first: Int, last: Int): TodoConnection
```

File: src/queries/getTodos.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        }
+      }
+    }
+  }
+}
+`;
+
+
+function GetTodos() {
+    const preloadedGetTodosQuery = preloadQuery(RelayEnvironment, TodoItemsQuery, {});
```
Comment (jevakallio) [inline src/queries/getTodos.js:27]: I don't know enough about Relay idiomatic patterns to be able to give feedback on this. Probably good to check with the Relay project maintainers

**Comment by SiddharthSham** [comment]
@jevakallio shall we open a PR upstream after performing the changes suggested, just so that we can get a review?

**Comment by harshil1712** [comment]
@SiddharthSham we don't have an upstream yet, for this repo.

**Comment by SiddharthSham** [comment]
@harshil1712 ah my bad. Let's directly link this PR?

**Comment by SiddharthSham** [comment]
@kassens could you please review this? 🙏

File: src/queries/getTodos.js (hunk 3 above)
Comment (kassens) [reply src/queries/getTodos.js:27]: React components with Suspense tend to render multiple times and rendering is potentially taking some time as well. What we prefer is loading a query as early as possible so we get the data as early as possible.

One simple option that's shown in the docs (search `preloadQuery` on [Step by Step Guide](https://relay.dev/docs/en/experimental/step-by-step#step-5-fetching-a-query-with-relay)) is to move the `preloadQuery` to the module level so it happens on module initialization time.

In more complicated apps that might navigate between different pages, ideally the router or navigation component could call this.

File: src/App.js [hunk 4]
```diff
@@ -1,7 +1,21 @@
...
 import "./App.css";
-import fetchGraphQL from "./fetchGraphQL";
+import {
+  RelayEnvironmentProvider,
+} from "react-relay/hooks";
+import RelayEnvironment from "./RelayEnvironment";
 
+<<<<<<< queries/getTodos
```
[the remark this answers is not in the document]
Comment (SiddharthSham) [reply src/App.js:8]: Hey, we've made the changes you suggested here in a new branch. Take a look at:

https://github.com/MLH-Fellowship/relay-examples/blob/mutations/updateTodoStatus/src/queries/getTodos.js

**Pull request merged**